pkg/ruler/rulespb/compat.go

```diff
@@ -28,12 +28,13 @@ func formattedRuleToProto(rls []rulefmt.RuleNode) []*RuleDesc {
 	rules := make([]*RuleDesc, len(rls))
 	for i := range rls {
 		rules[i] = &RuleDesc{
-			Expr:        rls[i].Expr.Value,
-			Record:      rls[i].Record.Value,
-			Alert:       rls[i].Alert.Value,
-			For:         time.Duration(rls[i].For),
-			Labels:      cortexpb.FromLabelsToLabelAdapters(labels.FromMap(rls[i].Labels)),
-			Annotations: cortexpb.FromLabelsToLabelAdapters(labels.FromMap(rls[i].Annotations)),
+			Expr:          rls[i].Expr.Value,
+			Record:        rls[i].Record.Value,
+			Alert:         rls[i].Alert.Value,
+			For:           time.Duration(rls[i].For),
+			KeepFiringFor: time.Duration(rls[i].KeepFiringFor),
```

Comment:
I think we also need to add this field in our API response.
https://github.com/prometheus/prometheus/blob/main/web/api/v1/api.go#L1348-L1364

Reply:
Same for the alert's KeepFiringSince https://github.com/prometheus/prometheus/blob/main/web/api/v1/api.go#L1216